src/capabilities.rs

```diff
@@ -194,6 +195,28 @@ impl SpecNewSessionParameters {
         Ok(())
     }
 
+    fn validate_no_proxy(value: &Json) -> WebDriverResult<()> {
+        match value.as_array() {
+            Some(hosts) => {
+                for host in hosts {
+                    match host.as_string() {
+                        Some(_) => {},
+                        None => return Err(WebDriverError::new(
+                            ErrorStatus::InvalidArgument,
+                            format!("{} was not a string", value)
+                        ))
+                    }
+                }
+            },
+            None => return Err(WebDriverError::new(
+                ErrorStatus::InvalidArgument,
+                format!("{} was not an array", value)
```

Comment:
Generally we should avoid passive language in error messages, and I think we could afford to be a tad more specific regarding which key this applies to.

I suggest `Expected noProxy to be an array, got: {}`.

(I know not every other error message here does that, but speaking from experience, it is almost impossible to decode them.)

Reply:
This would change how we report an invalid condition for this single key. I would suggest that I file a separate issue which can handle updating all items at once.

But I found a bug with the `String` check for individual entries, which should have `host` and not `value` as argument.

Reply:
To add I just used the wording and syntax from all the other checks...

Reply:
I filed https://github.com/mozilla/webdriver-rust/issues/122 for that.